azure-iot-device/tests/iothub/test_sync_clients.py

```diff
@@ -472,6 +477,88 @@ def test_no_message_in_inbox_nonblocking_mode(self, client, method_name):
         result = client.receive_method_request(method_name, block=False)
         assert result is None
 
+    @pytest.mark.it("Locks the client to API Receive Mode if the receive mode has not yet been set")
```

Comment:
observation: users can't mix api receive mode and handlers in the same app.  this is a great decision.  we need to make sure it's documented.

Reply:
Yeah, this will go in the docstrings and docs when the functionality is exposed

---
In reply to: [454524064](https://github.com/Azure/azure-iot-sdk-python/pull/620#discussion_r454524064) [](ancestors = 454524064)